### ISSUE TITLE
_uvc_stream_params_negotiated should skip dwMaxPayloadTransferSize check

### ISSUE THREAD

**Issue by fbzhong** [libuvc/libuvc#282]
I have a usb camera and the _uvc_stream_params_negotiated always return FALSE. According to the log:
```
14:17:41.083  D  [11098*stream.c:604/_uvc_stream_params_negotiated] Negotiation failed: required format 1, frame 3, size 800; actual format 1, frame 3, size 3072
```
The reason is required.dwMaxPayloadTransferSize is smaller than actual.dwMaxPayloadTransferSize.

According to the #273 , I think the dwMaxPayloadTransferSize should be ignored.

```c
static int _uvc_stream_params_negotiated(
  uvc_stream_ctrl_t *required,
  uvc_stream_ctrl_t *actual) {
    return required->bFormatIndex == actual->bFormatIndex &&
    required->bFrameIndex == actual->bFrameIndex;
}
```

**Comment by soonwonjang** [comment]
I agree with you. I experienced the same issue with the Kiyo Pro Ultra and Logitech MX Brio

**Comment by chelmuth** [comment]
Same with Logitech C920 PRO HD Webcam.

**Comment by fdegaulejac** [comment]
Same with my HD USB Video Capture Card from UGreen, works when removing the check on dwMaxPayloadTransferSize like suggested

**Comment by drinkcat** [comment]
(updated)

I'm not fully clear which of the 2 approaches in #277 or #292 is better (I'd lean towards an updated #277: https://github.com/drinkcat/libuvc/pull/17, but I'm not sure if that fixes the issue).
